### PR TITLE
feat: sync credential templates to managed agents

### PR DESCRIPTION
**Pull request by jparsai**
This PR is to add support for syncing Argo CD credential templates (repo-creds) from principal to
managed agents. Repository secret's project-scoped distribution logic is reused for repo-creds as well.
It also has an fix where deleting an AppProject on the principal did not
clean up the associated repository secrets from agents. Same is followed for repo-creds as well.

Fixes https://github.com/argoproj-labs/argocd-agent/issues/820

**How to test changes / Special notes to the reviewer**:
- Create an AppProject on the principal.
- Verify AppProject is synced to managed agent
- Create a repo-creds secret on the principal with `argocd.argoproj.io/secret-type: repo-creds` label and project field matching the AppProject name
- Verify the repo-creds secret is synced to managed agent but not to the autonomous agent
- Verify update/delete operations on principal are synced to managed agent.

Assisted by: Cursor

## Summary by CodeRabbit

* **New Features**
  * Sync repository credentials ("repo-creds") alongside repository secrets to managed agents and include them in source cache.

* **Bug Fixes**
  * Suppress noisy NotFound logs when projects are missing.
  * Trigger repository reconciliation and clean mappings on project deletion.
  * Return more specific errors when listing secrets by type fails.

* **Tests**
  * Added unit and E2E tests covering repo-creds acceptance, sync, AppProject delete/update reconciliation, and edge cases.

* **Refactor**
  * Unified handling of multiple repository secret types across listing, sync, and cleanup flows.

### CODE REVIEW

File: test/e2e/repocreds_test.go [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+	requires.NoError(suite.PrincipalClient.Create(suite.Ctx, &repoCreds, metav1.CreateOptions{}))
+
+	key := fixture.ToNamespacedName(&repoCreds)
+	requires.Eventually(func() bool {
+		tempRepoCreds := corev1.Secret{}
+		err := suite.ManagedAgentClient.Get(suite.Ctx, key, &tempRepoCreds, metav1.GetOptions{})
+		return err == nil
+	}, 60*time.Second, 1*time.Second, "repo-creds should be synced to the managed-agent")
+
+	return repoCreds
```
[the remark this answers is not in the document]
Comment (jparsai) [reply test/e2e/repocreds_test.go:98]: I don't think this is needed. There are no tests that create repo/repo-creds secrets without label also `CleanUp` function removes all secrets that have label. So we may not face this issue at all.

File: internal/backend/kubernetes/repository/kubernetes.go [hunk 2]
```diff
@@ -73,7 +73,12 @@ func WithLabelSelector(labelSelector string) KubernetesBackendOption {
 }
 
 func (be *KubernetesBackend) List(ctx context.Context, selector backend.RepositorySelector) ([]corev1.Secret, error) {
-	labelSelector := common.LabelKeySecretType + "=" + common.LabelValueSecretTypeRepository
+	var labelSelector string
+	if secretType, ok := selector.Labels[common.LabelKeySecretType]; ok {
+		labelSelector = common.LabelKeySecretType + "=" + secretType
+	} else {
+		labelSelector = common.LabelKeySecretType + "=" + common.LabelValueSecretTypeRepository
+	}
```
Comment (jgwest) [inline internal/backend/kubernetes/repository/kubernetes.go]: IMHO, let's just return an error in the `!ok` case: AFAICT, all callers of the function are currently setting the labelSelector to include a secret type. WDYT?

Comment (jparsai) [reply internal/backend/kubernetes/repository/kubernetes.go]: Thanks @jgwest, it is updated now, PTAL
